src/core/readyException.js

```diff
@@ -0,0 +1,13 @@
+define( [
+	"../core"
+], function( jQuery ) {
+
+"use strict";
+
+jQuery.readyException = function( error ) {
+	window.setTimeout( function() {
+		throw error;
+	} );
```

Comment:
I forgot @gibson042 wanted to have this reject as well so that a more general potential unhandled rejections mechanism would catch it. The code would be:
```js
jQuery.readyException = function( error ) {
	window.setTimeout( function() {
		throw error;
	} );
	throw error;
};
```

I have doubts about the need for that, though - the error thrown in a timeout would fire `window.onerror` anyway, do we need both mechanism to work here? Usually an error is either synchronous and handled by `window.onerror` or asynchronous and handled by the other mechanism.

cc @dmethvin as you already LGTM'd this PR. :)